candle-examples/examples/moondream/main.rs

```diff
@@ -64,6 +72,10 @@ impl TextGeneration {
         let mut tokens = tokens.get_ids().to_vec();
         let mut generated_tokens = 0usize;
 
+        let bos_token = match self.tokenizer.get_vocab(true).get("<|endoftext|>") {
```

Comment:
This feels a bit weird for a bos token. Maybe add a comment if that's expected so as to mention that it's actually what the moondream2 tokenizer is using (maybe with a pointer to some json config or to the code?).

Reply:
I'm also a tad confused about this. [This](https://huggingface.co/vikhyatk/moondream2/blob/main/special_tokens_map.json) json file specified all tokens as "<|endoftext|>" which is akin to Phi which uses the same token for EOS and BOS. 

However, the Python moondream implementation uses "<END>" as EOS token. Additionally, "<END>" doesn't seem to be working in candle, but "END" does work. I looked at the tokenizer and "<END>" doesn't seem to be a token either. Maybe I'm missing something.